rest_framework/views.py

```diff
@@ -275,7 +275,11 @@ def get_permissions(self):
         """
         Instantiates and returns the list of permissions that this view requires.
         """
-        return [permission() for permission in self.permission_classes]
+        try:
```

Comment:
can you elaborate what this change will help to achieve?

Reply:
For each actions (list, retrieve, create, ...) in one api you can use different permissions. For example, in ComponentView, list action permission is "AllowAny" and create action is "IsAdminUser". This means anyone can get Component objects but just admin user can create object.